kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala

```diff
@@ -31,8 +30,8 @@ private[client] trait Creatable[F[_], Resource <: { def metadata: Option[ObjectM
         EnrichedStatus[F]
       )
 
-  def createOrUpdate(resource: Resource): F[Status] = {
-    val fullResourceUri = config.server.resolve(resourceUri) / resource.metadata.get.name.get
+  def createOrUpdate(resource: Resource)(implicit F: Sync[F]): F[Status] = {
+    val fullResourceUri = config.server.resolve(resourceUri) / resource.metadata.flatMap(_.name).get
```

Comment:
Those `get` are not good but the `flatMap` is not improving the status quo.
I feel like breaking soon rather than later is better.

Reply:
i would rather deprecate this and include the name into the signature

Reply:
You mean return an `Option`?

Reply:
`createOrUpdate(resource: Resource, name: String)`

Reply:
or even returning an option, if we cant find the name,
it can be an `F[Option[Status]`, OptionT[F, Status], or Option[F[Status]

Reply:
I see. Name is mandatory in `Resource` anyway. Here we have an `Option` because it's not specified as mandatory in the swagger and we derive case classes from it. I which it was.

Reply:
I left this as is. I dont like the Option.get stuff, but there is really no way of handling this statically.